src/overlay/OverlayEnvelopeHandler.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+// Copyright 2018 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "overlay/EnvelopeHandler.h"
+
+namespace stellar
+{
+class OverlayEnvelopeHandler : public EnvelopeHandler
+{
+  public:
+    explicit OverlayEnvelopeHandler(Application& app);
+    ~OverlayEnvelopeHandler() = default;
+
+    void setValidRange(uint32_t min, uint32_t max) override;
+    EnvelopeStatus envelope(Peer::pointer,
+                            SCPEnvelope const& envelope) override;
+
+    void getQuorumSet(Peer::pointer peer, Hash const& hash) override;
+    std::set<SCPEnvelope> quorumSet(Peer::pointer peer,
```

Comment:
same thing wrt naming, `quorumSet` and `txSet` are not good names; maybe prefix with `recv`